sources/nodeUtils.ts

```diff
@@ -4,6 +4,10 @@ export interface NodeError extends Error {
   code: string;
 }
 
+export function isExistError(err: NodeError) {
```

Comment:
We can use this as an opportunity to reduce the number of `as` in the codebase
```suggestion
export function isNodeError(err: unknown): err isNodeError {
  return !!err?.code;
}

export function isExistError(err: NodeError) {
```